test/models/archived_patient_test.rb

```diff
@@ -0,0 +1,90 @@
+                                other_contact: 'Yolo'
+
+    @patient.calls.create attributes_for(:call, created_by: @user, status: 'Reached patient')
+    create_language_config
+    @archived_patient = create :archived_patient, line: 'DC',
+                                initial_call_date: 400.days.ago,
+                                created_by_id: @user.id
+  end
+
+  describe 'validations' do
+    it 'should build' do
+      assert @archived_patient.valid?
+    end
+
+    it 'requires a line' do
+      @archived_patient.line = nil
+      refute @archived_patient.valid?
+    end
+
+    it 'requires an initial call date' do
+      @archived_patient.initial_call_date = nil
+      refute @archived_patient.valid?
+    end
+
+    it 'requires a logged creating user' do
+      @archived_patient.created_by_id = nil
+      refute @archived_patient.valid?
+    end
+  end
+
+
+  describe 'Archiving Methods' do
+    before do
+      @patient_old = create :patient, primary_phone: '222-222-3333',
+                                      other_phone: '222-222-4444',
+                                      initial_call_date: 600.days.ago
+      @patient_fulfilled = create :patient, primary_phone: '222-222-3334',
+                                   other_phone: '222-222-4443',
+                                   initial_call_date: 310.days.ago
+      @patient_fulfilled.fulfillment.fulfilled = true
+      @patient_fulfilled.fulfillment.date_of_check = 300.days.ago
+      @patient_fulfilled.fulfillment.procedure_date = 290.days.ago
+      @patient_fulfilled.fulfillment.check_number = '123'
+
+      @patient_fulfilled2 = create :patient, primary_phone: '222-212-3334',
+                                   other_phone: '222-222-4443',
+                                   initial_call_date: 310.days.ago
+      @patient_fulfilled2.fulfillment.fulfilled = true
+      @patient_fulfilled2.fulfillment.date_of_check = 300.days.ago
+      @patient_fulfilled2.fulfillment.procedure_date = 290.days.ago
+      @patient_fulfilled2.fulfillment.check_number = '123'
+      @patient4 = create :patient, primary_phone: '222-222-3336',
+                                   other_phone: '222-222-4441',
+                                   line: 'DC',
+                                   city: 'Washington',
+                                   race_ethnicity: 'Asian',
+                                   initial_call_date: 16.days.ago,
+                                   appointment_date: 6.days.ago
+    end
+
+    it 'Should create an archive copy of a patient' do
+      @archived_patient = ArchivedPatient.convert_patient(@patient4)
+      @archived_patient.save!
+      assert_equal @archived_patient.line, @patient4.line
+      assert_equal @archived_patient.city, @patient4.city
+      assert_equal @archived_patient.race_ethnicity, @patient4.race_ethnicity
+      assert_equal @archived_patient.appointment_date, @patient4.appointment_date
```

Comment:
We should probably test the `destroy!` part of this process too, I think?

Reply:
threw in checks for Patient.count decrementing.

Reply:
💯 this is pretty much exactly what I had in mind!